src/gui/editorwidgets/qgsvaluemapconfigdlg.cpp

```diff
@@ -345,3 +352,39 @@ void QgsValueMapConfigDlg::loadMapFromCSV( const QString &filePath )
 
   updateMap( map, false );
 }
+
+bool QgsValueMapConfigDlg::validateKeys() const
+{
+  bool allValid = true;
+  if ( !tableWidget )
+    return false;
+  for ( int i = 0; i < tableWidget->rowCount() - 1; i++ )
+  {
+    QTableWidgetItem *ki = tableWidget->item( i, 0 );
```

Comment:
can you use a smart pointer here?

Reply:
I've been experimenting with implementing smart pointers, and now whenever the smart pointer is used, the content of the cell just get emptied and the color change on invalid values causes crashes. I can commit those changes if you want but I don't know how to prevent that and how the smart pointer would improve things...

Reply:
@lbartoletti I've checked more about unique ptr and the deletion of the object is intended, that explain why the text and background get deleted after validation. Did you had in mind a shared pointer instead? I've tried with Qt smart pointers and  they seem to be a hassle with QTableWidgetItem, I'm not sure why advantage they would provide as the QTableWidgetItem still need to exist, its text and background are used, but the object still need to exist within the widget, unsure why changing pointer types would be needed.

Standard pointers seem to have always been used with QTableWidgetItem in the codebase without much issues.